src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -77,25 +76,32 @@ public static void setDatesToRepoConfigs(
      * Merges a {@code RepoConfiguration} from {@code repoConfigs} with another from {@code authorConfigs}
      * if {@code location} and {@code branch} matches.
      */
-    public static void merge(List<RepoConfiguration> repoConfigs, List<RepoConfiguration> authorConfigs) {
-        for (RepoConfiguration authorConfig : authorConfigs) {
-            if (authorConfig.location.isEmpty()) {
+    public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfiguration> authorConfigs) {
+        for (AuthorConfiguration authorConfig : authorConfigs) {
+            if (authorConfig.getLocation().isEmpty()) {
                 for (RepoConfiguration repoConfig : repoConfigs) {
                     repoConfig.addAuthors(authorConfig.getAuthorList());
                 }
                 continue;
             }
 
-            int index = repoConfigs.indexOf(authorConfig);
+            RepoConfiguration matchingRepoConfig = null;
 
-            if (index == -1) {
+            for (RepoConfiguration repoConfig: repoConfigs) {
+                if (repoConfig.getLocation().equals(authorConfig.getLocation())
+                        && repoConfig.getBranch().equals(authorConfig.getBranch())) {
+                    matchingRepoConfig = repoConfig;
```

Comment:
perhaps extract this entire chunk as a static method that returns `matchingConfig` from List of `RepoConfiguration`?